## Procedures to evaluate simulated data
### Simulated data generated by Mason
1. Run mason and select the reads from sequences with valid taxonomy information. This generate 1 million paired-end data
```
mason illumina -pi 0 -pd 0 -pmms 2.5 -s 17 -N 2000000 -n 100 -mp -sq  -hs 0 -hi 0 -i -o p_sim_2M.fq p_genomes.fa

perl ExtractValidSimReads.pl p_sim_2M_1.fq p_seqid.map nodes.dmp 1000000 > p_sim_1M_1.fq
perl ExtractValidSimReads.pl p_sim_2M_2.fq p_seqid.map nodes.dmp 1000000 > p_sim_1M_2.fq
```

2. Generate the truth data

```
perl GetTruthTable.pl p_sim_1M_1.fq p_seqid.map > sim_truth.out
```

3. Run classifier

Centrifuger: centrifuger -t 8 -x cfr_p -1 $r1 -2 $r2 > cfr_class.out

Centrifuge:
```
centrifuge -no-abundance -p 8 -x centrifuge_p -1 $r1 -2 $r2 > centrifuge_class_raw.out
centrifuge-promote centrifuge_p centrifuge_class_raw.out lca > centrifuge_class.out
```

Kraken2: ~/Softwares/kraken2/kraken2 --paired --db kraken_p_default --threads 8 $r1 $r2 > kraken_class.out 

Then we shall select the three columns, read_id, taxonomy_id and score as the final output for the evaluation script, e.g. "cut -f1,3,4 cfr_class.out | grep -v readID > cfr_3c.out".

4. Run evaluation, using centrifuger's transformed output as the example.
```
python3 sim_evaluate.py --truth sim_truth.out --tree nodes.dmp -c cfr_3c.out
```

### CAMI2's simulated data
Generate CAMI2's truth table
```
zcat reads_mapping.tsv.gz | grep -v "^#" | awk 'NR%2==1' | cut -f1,3 | sed "s/\/1//g" > sim_truth.out
```

The running command and the evaluation procedure is the same as the evaluation for Mason-generated simulated data.


### Centrifuger reassign
```
centrifuger -t 8 -x cfr_p -1 $r1 -2 $r2 -k 5 > cfr_class.out
centrifuger-kreport -x cfr_p cfr_class.out > cfr_kreport.out
../other_scripts/centrifuger-reassign cfr_p cfr_class.out cfr_kreport.out > cfr_reassign.out

```
